app/full_text_search.py

```diff
@@ -42,25 +73,40 @@ def search(repo, query_txt, max_results):
     results = []
     if not query_txt:
         return results
-    index = appengine_search.Index(name=PERSON_FULL_TEXT_INDEX_NAME)
-    query_txt += ' AND (repo: ' + repo + ')'
+
+    # escape double quote for enclose each query_txt
+    query_txt = re.sub('"', '', query_txt)
+
+    person_location_index = appengine_search.Index(
+        name=PERSON_LOCATION_FULL_TEXT_INDEX_NAME)
     options = appengine_search.QueryOptions(
         limit=max_results,
-        returned_fields=['record_id'])
-    index_results = index.search(appengine_search.Query(
-        query_string=query_txt, options=options))
-    for document in index_results:
-        id = document.fields[0].value
-        result = model.Person.get(repo, id, filter_expired=True)
-        if result:
-            results.append(result)
-    return results
+        returned_fields=['record_id', 'names'])
+
+    and_query = enclose_in_double_quotes(query_txt) + ' AND (repo: ' + repo + ')'
+    person_location_index_results = person_location_index.search(
+        appengine_search.Query(
+            query_string=and_query, options=options))
+    index_results = []
+    regexp = make_or_regexp(query_txt)
+    for document in person_location_index_results:
+        for field in document.fields:
+            if field.name == 'names':
+                names = field.value
+            if field.name == 'record_id':
+                id = field.value
 
+        if regexp.search(names):
+            index_results.append(id)
+
+    return [model.Person.get(repo, id, filter_expired=True)
```

Comment:
This code can call model.Person.get() twice for a same record, which is bad. Can you avoid it? You may give up using [... for ... in ...] syntax.